language/stdlib/modules/validator_config.move

```diff
@@ -1,156 +1,183 @@
+// Error codes:
+// 1100 -> OPERATOR_ACCOUNT_DOES_NOT_EXIST
+// 1101 -> INVALID_TRANSACTION_SENDER
+// 1102 -> VALIDATOR_NOT_CERTIFIED
+// 1103 -> VALIDATOR_OPERATOR_NOT_CERTIFIED
+// 1104 -> VALIDATOR_CONFIG_IS_NOT_SET
+// 1105 -> VALIDATOR_OPERATOR_IS_NOT_SET
 address 0x0 {
 
 module ValidatorConfig {
-    use 0x0::LibraAccount;
+    use 0x0::Association;
     use 0x0::Option;
     use 0x0::Transaction;
+    use 0x0::ValidatorOperatorConfig;
 
     struct Config {
         consensus_pubkey: vector<u8>,
         // TODO(philiphayes): restructure
+        //   1) remove validator_network_signing_pubkey
+        //   2) make validator_network_address[es] of type vector<vector<u8>>,
+        //   3) make fullnodes_network_address[es] of type vector<vector<u8>>,
+        //   4) remove validator_network_identity_pubkey
+        //   5) remove full_node_network_identity_pubkey
         validator_network_signing_pubkey: vector<u8>,
         validator_network_identity_pubkey: vector<u8>,
         validator_network_address: vector<u8>,
         full_node_network_identity_pubkey: vector<u8>,
         full_node_network_address: vector<u8>,
     }
 
-    // A current or prospective validator should publish one of these under their accounts.
-    // If delegation is enabled, only the delegated_account can change the config.
-    // If delegation is disabled, only the owner of this resource can change the config.
-    // The entity that can change the config is called a "validator operator".
-    // The owner of this resource can enable delegation and set delegated account.
-    // The owner can also disable delegation at any time.
     resource struct T {
-        config: Config,
-        delegated_account: Option::T<address>,
+        // the flag is set by the Association
+        is_certified: bool,
+        // set and rotated by the operator_account
+        config: Option::T<Config>,
+        operator_account: Option::T<address>,
     }
 
     // TODO(valerini): add events here
 
-    // Returns true if addr has a published ValidatorConfig::T resource
-    public fun has(addr: address): bool {
-        exists<T>(addr)
+    ///////////////////////////////////////////////////////////////////////////
+    // Validator setup methods
+    ///////////////////////////////////////////////////////////////////////////
+
+    // This function instantiates a
+    // ValidatorConfig resourse under validator_address
+    public fun initialize(validator_address: address) {
```

Comment:
since eventually this will be part of the account create process, my recommendation would be to have the rotate_operator try to instantiate this object if it doesn't exist, if it does exist, then it just rotates the operator.